plugins/multilevel-linear-view/src/LinearGenomeMultilevelView/model.ts

```diff
@@ -0,0 +1,451 @@
+            const [refName, start, end] = inputs
+            if (
+              `${e}`.match(/Unknown reference sequence/) &&
+              Number.isInteger(+start) &&
+              Number.isInteger(+end)
+            ) {
+              parsedLocStrings = [
+                parseLocString(refName + ':' + start + '..' + end, ref =>
+                  isValidRefName(ref, assemblyName),
+                ),
+              ]
+            } else {
+              throw e
+            }
+          }
+
+          const locations = parsedLocStrings.map(region => {
+            const asmName = region.assemblyName || assemblyName
+            const asm = assemblyManager.get(asmName)
+            const { refName } = region
+            if (!asm) {
+              throw new Error(`assembly ${asmName} not found`)
+            }
+            const { regions } = asm
+            if (!regions) {
+              throw new Error(`regions not loaded yet for ${asmName}`)
+            }
+            const canonicalRefName = asm.getCanonicalRefName(region.refName)
+            if (!canonicalRefName) {
+              throw new Error(
+                `Could not find refName ${refName} in ${asm.name}`,
+              )
+            }
+            const parentRegion = regions.find(
+              region => region.refName === canonicalRefName,
+            )
+            if (!parentRegion) {
+              throw new Error(`Could not find refName ${refName} in ${asmName}`)
+            }
+
+            return {
+              ...region,
+              assemblyName: asmName,
+              parentRegion,
+            }
+          })
+
+          if (locations.length === 1) {
+            const loc = locations[0]
+            // @ts-ignore
+            self.setDisplayedRegions([
+              { reversed: loc.reversed, ...loc.parentRegion },
+            ])
+            const { start, end, parentRegion } = loc
+            // @ts-ignore
+            self.navTo({
+              ...loc,
+              start: clamp(start ?? 0, 0, parentRegion.end),
+              end: clamp(end ?? parentRegion.end, 0, parentRegion.end),
+            })
+          } else {
+            // @ts-ignore
+            self.setDisplayedRegions(
+              // @ts-ignore
+              locations.map(r => (r.start === undefined ? r.parentRegion : r)),
+            )
+            // @ts-ignore
+            self.showAllRegions()
+          }
+        },
+        moveIfAnchor(leftOffset: number, rightOffset: number) {
+          if (self.isAnchor) {
+            // @ts-ignore
+            self.moveTo(leftOffset, rightOffset)
+          }
+        },
+        closeView() {
+          const parent = getContainingView(self)
+          parent.removeView(self)
+        },
+        addView(isAbove: boolean) {
+          const parent = getContainingView(self)
+          parent.addView(isAbove, self)
+        },
+      }))
+      .views(self => ({
+        menuItems(): MenuItem[] {
```

Comment:
The reason I duplicated the menu instead of super'ing it was because when appending to a super menu the "checked" property being triggered is not observed for some reason...

To be clear,

If I append the following to a `superMenuItems` array:

```js
          {
            label: 'Show controls',
            icon: VisibilityIcon,
            type: 'checkbox',
            checked: !self.hideControls,
            onClick: self.toggleControls,
            disabled: !self.isVisible || self.isAnchor,
          },
```

clicking on the menu item does execute the action of 'show/hide controls', but the checkbox being toggled is not observed, i.e. clicking that menu item results in the box remaining unchecked, but the controls being shown.

Any ideas for getting around this?

Reply:
This is roughly what the .views() looks like utilizing `superMenuItems` approach:

```js
      .views(self => {
        // @ts-ignore
        const { menuItems: superMenuItems } = self
        const superMenuItemsArray = superMenuItems()

        const index = superMenuItemsArray.findIndex(
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          (item: any) => item.label === 'Return to import form',
        )

        superMenuItemsArray.splice(index, 1)

        const addRemoveMenuItems = [
          !self.isAnchor && !self.isOverview
            ? {
                label: 'Remove view',
                icon: CloseIcon,
                onClick: self.closeView,
              }
            : {
                label: 'This view cannot be removed',
                icon: CloseIcon,
                disabled: true,
                onClick: () => {},
              },
          {
            label: 'Add neighbouring view',
            icon: AddIcon,
            subMenu: [
              {
                label: 'Add view above',
                icon: VerticalAlignTopIcon,
                onClick: () => {
                  self.addView(true)
                },
              },
              {
                label: 'Add view below',
                icon: VerticalAlignBottomIcon,
                onClick: () => {
                  self.addView(false)
                },
              },
            ],
          },
        ]

        superMenuItemsArray.splice(2, 0, ...addRemoveMenuItems)

        const controlsHideMenuItems = [
          {
            label: 'Show controls',
            icon: VisibilityIcon,
            type: 'checkbox',
            checked: !self.hideControls,
            onClick: self.toggleControls,
            disabled: !self.isVisible || self.isAnchor,
          },
          {
            label: 'Hide view',
            icon: VisibilityIcon,
            type: 'checkbox',
            checked: !self.isVisible,
            onClick: self.toggleVisible,
          },
        ]

        superMenuItemsArray.splice(12, 0, ...controlsHideMenuItems)

        return {
          menuItems(): MenuItem[] {
            return superMenuItemsArray
          },
        }
      })
```

Reply:
`superMenuItems()` needs to get called within the returned function, so it should work if you have something like this:

```ts
      .views(self => {
        // @ts-ignore
        const { menuItems: superMenuItems } = self
        return {
          menuItems(): MenuItem[] {
            const superMenuItemsArray: MenuItem[] = superMenuItems()

            const index = superMenuItemsArray.findIndex(
              // eslint-disable-next-line @typescript-eslint/no-explicit-any
              (item: any) => item.label === 'Return to import form',
            )

            superMenuItemsArray.splice(index, 1)

            const addRemoveMenuItems = [
              !self.isAnchor && !self.isOverview
                ? {
                    label: 'Remove view',
                    icon: CloseIcon,
                    onClick: self.closeView,
                  }
                : {
                    label: 'This view cannot be removed',
                    icon: CloseIcon,
                    disabled: true,
                    onClick: () => {},
                  },
              {
                label: 'Add neighbouring view',
                icon: AddIcon,
                subMenu: [
                  {
                    label: 'Add view above',
                    icon: VerticalAlignTopIcon,
                    onClick: () => {
                      self.addView(true)
                    },
                  },
                  {
                    label: 'Add view below',
                    icon: VerticalAlignBottomIcon,
                    onClick: () => {
                      self.addView(false)
                    },
                  },
                ],
              },
            ]

            superMenuItemsArray.splice(2, 0, ...addRemoveMenuItems)
            console.log(self.isVisible)

            const controlsHideMenuItems: MenuItem[] = [
              {
                label: 'Show controls',
                icon: VisibilityIcon,
                type: 'checkbox',
                checked: !self.hideControls,
                onClick: self.toggleControls,
                disabled: !self.isVisible || self.isAnchor,
              },
              {
                label: 'Hide view',
                icon: VisibilityIcon,
                type: 'checkbox',
                checked: !self.isVisible,
                onClick: self.toggleVisible,
              },
            ]

            superMenuItemsArray.splice(12, 0, ...controlsHideMenuItems)

            return superMenuItemsArray
          },
        }
      })
```